src/System.Net.Sockets/src/System/Net/Sockets/Socket.Tasks.cs

```diff
@@ -424,10 +487,37 @@ private static void ValidateBuffersList(IList<ArraySegment<byte>> buffers)
         }
 
         /// <summary>Completes the SocketAsyncEventArg's Task with the result of the send or receive, and returns it to the specified pool.</summary>
-        private static void CompleteSendReceive(Int32TaskSocketAsyncEventArgs saea, bool isReceive)
+        private static void CompleteAccept(Socket s, TaskSocketAsyncEventArgs<Socket> saea)
+        {
+            // Pull the relevant state off of the SAEA
+            SocketError error = saea.SocketError;
+            Socket acceptSocket = saea.AcceptSocket;
+
+            // Synchronize with the initiating thread. If the synchronous caller already got what
+            // it needs from the SAEA, then we can return it to the pool now. Otherwise, it'll be
+            // responsible for returning it once it's gotten what it needs from it.
+            bool responsibleForReturningToPool;
+            AsyncTaskMethodBuilder<Socket> builder = saea.GetCompletionResponsibility(out responsibleForReturningToPool);
```

Comment:
I like it for cases like:
```C#
if (blah.TryWhatever(out T foo))
{
    Use(foo);
}
```
but I don't like it for the case highlighted in the code.  With declarations in such positions, when I'm reading through code and I want to see what type something is, previously I could essentially scan the left side of all lines in the code... now I need to basically scan all of the code.